MaxText/layers/attentions.py

```diff
@@ -759,9 +762,6 @@ def kv_cache_autoregressive(
     assert cached_ar_key_var[0].value.shape == self.cached_kv_shape((batch, self.max_target_length - self.max_prefill_predict_length, heads, kv_head_size), self.ar_key_axis_order)
     assert cached_ar_value_var[0].value.shape == self.cached_kv_shape((batch, self.max_target_length - self.max_prefill_predict_length, heads, kv_head_size), self.ar_value_axis_order)
 
-    key = nn.with_logical_constraint(key, (BATCH, LENGTH, HEAD, D_KV))
-    value = nn.with_logical_constraint(value, (BATCH, LENGTH, HEAD, D_KV))
-
```

Comment:
Does removing the logical constraint have any impact? I know in some cases, compiler is not able to propagate the sharding, thus hard coded nn.with_logical_constraint is in places in some cases.

Reply:
There are already annotated in https://github.com/google/maxtext/blob/main/MaxText/layers/attentions.py#L1057-L1062.

Reply:
Instead of removing them could you replace them with the more specific axis names?

Reply:
If they are already annotated in the https://github.com/google/maxtext/blob/main/MaxText/layers/attentions.py#L1057-L1062, I don't think we need to annotate it again. (just like we don't annotate it in the kv_cache_prefill, I think we don't need to annotate it in the kv_cache_autoregressive as well)

Reply:
I think these two lines are safe to be deleted given https://github.com/google/maxtext/blob/main/MaxText/layers/attentions.py#L1057-L1062